enhancements/network/ip-interface-selection.yaml

```diff
@@ -0,0 +1,302 @@
+a feature allowing deployers to specify that they want cluster traffic on
+eth0, external traffic on eth1, and storage traffic on eth2, that feature
+would simply need to appropriately populate the KUBELET_NODEIP_HINT file that
+would be directly created by the deployer in the current design. By providing
+a common interface to configure host-networked services, this should actually
+simplify any such future enhancements.
+
+### Risks and Mitigations
+
+- There is some risk to changing the order of critical system services like
+  nodeip-configuration and configure-ovs. This will not affect deployments
+  that do not use OVNKubernetes as their CNI, but since we intend that to be
+  the default going forward it is a significant concern.
+
+  We intend to mitigate this risk by first merging the order change without
+  any additional changes included in this design. This way, if any races
+  between services are found once the change is being tested more broadly
+  it will be easy to revert.
+
+  We will also test the ordering change as much as possible before merging
+  it, but it's unlikely we can exercise it to the same degree that running
+  across hundreds of CI jobs per day will.
+
+- Currently all host services are expected to listen on the same IP and
+  interface. If at some point in the future we need host services listening
+  on multiple different interfaces, this may not work. However, because we
+  are centralizing all IP selection logic in nodeip-configuration, it should
+  be possible to extend that to handle multiple interfaces if necessary.
+
+### Drawbacks
+
+This design only considers host networking, and it's likely that in the future
+we will want a broader feature that provides an interface to configure pod
+traffic routing as well. However, if/when such a feature is implemented it
+should be able to use the same configuration interface for host services
+that deployers would use directly after this is implemented.
+
+Additionally, there are already ways to [implement traffic steering for pod
+networking.](https://youtu.be/EpbUWwjadYM) We may at some point want to
+integrate them more closely, but host networking is currently a much bigger
+pain point and worth addressing on its own.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Currently this only applies to UPI clusters deployed using platform None.
+  Do we need something similar for IPI?
+
+- In UPI deployments it is also possible to set the Node IP by manually writing
+  configuration files for Kubelet and CRIO. Trying to look for all possible
+  ways a user may have configured those services seems complex and error-prone.
+  Can we just require them to use this mechanism if they want custom IP
+  selection?
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
```

Comment:
We need tests for both configure-ovs as well as for this entire mechanism, ideally covering some nasty corner cases. Right now, due to the absence of tests, we are constantly at risk of introducing further issues wwhile trying to fix something.